test/config/test/console.test.ts

```diff
@@ -21,3 +22,44 @@ test.each(['threads', 'vmThreads'] as const)(`disable intercept pool=%s`, async
   const call = spy.mock.lastCall![0]
   expect(call.toString()).toBe('__test_console__\n')
 })
+
+test('group synchronous console logs', async () => {
+  const logs: UserConsoleLog[] = []
+  await runVitest({
+    root: './fixtures/console-batch',
+    reporters: [
+      'default',
+      {
+        onUserConsoleLog(log) {
+          logs.push(log)
+        },
```

Comment:
Instead of asserting `onUserConsoleLog`, could we directly check the `stdout` of the process? I feel like that would provide more confidence of the fix from user's perspective.

Something like:

```ts
test('group synchronous console logs', async () => {
  const { stdout } = await runVitest({
    root: './fixtures/console-batch',
    reporters: ['default'],
  })

  const logs = stdout
    .split('\n')
    .filter(row => row.length === 0 || row.startsWith('stdout') || row.startsWith('['))
    .join('\n')
    .trim()

  expect(logs).toMatchInlineSnapshot(...)
```

Reply:
Nice trick! I didn't think there's a robust way to test stdout directly :open_mouth: